### PR TITLE
Fixed bugs and cmake build for Windows

### PR DESCRIPTION
**Pull request by alkama**
- CMake-based build system now works for OSX, Windows (and probably Linux).
- Incorrect use of "wcscpy_s" and "swprintf_s". The expected size is in number of characters, not bytes. It used to cause stack corruption.
- Missing header.

### CODE REVIEW

File: ogl_editor/src/windows/RocketWindow.c [hunk 1]
```diff
@@ -210,17 +210,17 @@ static void formatName(wchar_t* outName, int keyMod, int key, const wchar_t* nam
...
-			case EMGUI_KEY_ARROW_UP: wcscpy_s(keyName, sizeof(keyName), L"Up"); break;
-			case EMGUI_KEY_ARROW_RIGHT: wcscpy_s(keyName, sizeof(keyName), L"Right"); break;
-			case EMGUI_KEY_ARROW_LEFT: wcscpy_s(keyName, sizeof(keyName), L"Left"); break;
-			case EMGUI_KEY_ESC: wcscpy_s(keyName, sizeof(keyName), L"ESC"); break;
-			case EMGUI_KEY_TAB: wcscpy_s(keyName, sizeof(keyName), L"TAB"); break;
-			case EMGUI_KEY_BACKSPACE: wcscpy_s(keyName, sizeof(keyName), L"Delete"); break;
-			case EMGUI_KEY_ENTER: wcscpy_s(keyName, sizeof(keyName), L"Enter"); break;
-			case EMGUI_KEY_SPACE: wcscpy_s(keyName, sizeof(keyName), L"Space"); break;
-			case EMGUI_KEY_PAGE_UP: wcscpy_s(keyName, sizeof(keyName), L"Page Up"); break;
-			case EMGUI_KEY_PAGE_DOWN: wcscpy_s(keyName, sizeof(keyName), L"Page Down"); break;
```
Comment (emoon) [inline ogl_editor/src/windows/RocketWindow.c]: I would rather have these numbers calculated instead of having a hard-code value.

**Comment by alkama** [comment]
You're right :+1: 
At first I thought it was kind of ugly to resort to using sizeof on an element containing garbage. That's why I favored using the same hardcoded value that the array uses in its local definition.
We could also probably create a length variable and calc that size once, but well, at least that solution is close to the original one :)

**Comment by emoon** [comment]
I was more thinking of doing something like size = sizeof(text) * something that is correct than just using 64 here.

**Comment by alkama** [comment]
The new version still dont fit?
Anyway sizeof in this case will always involve ugly things like:
* `sizeof(keyName)/sizeof(keyName[0])`
* `sizeof(keyName)/sizeof(wchar_t)`
* `sizeof(keyName)/4` :)
* `sizeof(keyName)*0.25` YOLO!!!!

I think there is a `_countof` but it's visual studio only, and c++.

**Comment by emoon** [comment]
https://github.com/emoon/rocket/blob/master/ogl_editor/emgui/include/emgui/Types.h#L94 :)

**Comment by alkama** [comment]
THIS is why "you rule" :)

**Comment by emoon** [comment]
haha ok :)

**Comment by emoon** [comment]
Thanks for the change! Looks good now :)

**Pull request merged**